runner.go

```diff
@@ -442,19 +452,29 @@ func (runner *Runner) startWorker(req startReq) error {
 		return nil
 	}
 	info := runner.workers[req.id]
-	if info == nil {
-		runner.mu.Lock()
-		defer runner.mu.Unlock()
-		runner.workers[req.id] = &workerInfo{
-			start:        req.start,
-			restartDelay: runner.params.RestartDelay,
-			started:      runner.params.Clock.Now().UTC(),
-			done:         make(chan struct{}, 1),
-		}
-		go runner.runWorker(0, req.id, req.start)
-		return nil
+	if info != nil {
+		return errors.AlreadyExistsf("worker %q", req.id)
 	}
-	return errors.AlreadyExistsf("worker %q", req.id)
+
+	labels := pprof.Labels(
+		"worker-id", req.id,
+		"worker-name", workerFuncName(req.start),
```

Comment:
I also wish runtime provided a cleaner way. It is a bit of a shame to see both 'workerFuncName' and 'funcName'.
I don't have a good `utilities` package to put it in, though. (As all misc/utility packages just accrete until they are unusable.)

Reply:
Agreed 100%